File: java/src/test/java/dependency/injection/java/t01integration/dependency/TestSetter.java
package dependency.injection.java.t01integration.dependency;

// internal
import dependency.injection.java.dependency.Setter;
import dependency.injection.java.dependency.Fibonacci;
import dependency.injection.java.dependency.Square;

// external
import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;


public class TestSetter {
    
    @Test
    public void test_integration__doAll__validData() {
        var fibonacci = new Fibonacci();
        var square = new Square();
        var setter = new Setter();
        setter.setFibonacci(fibonacci);
        setter.setSquare(square);

        setter.doAll(10);

        assertEquals(true, true);
    }
}

File: python/tests/t01integration/dependency/test_setter.py
# internal
from dependency.fibonacci import Fibonacci
from dependency.square import Square
from dependency.setter import Setter


def test__integration__do_all__valid_data():
    fibonacci = Fibonacci()
    square = Square()
    setter = Setter()
    setter.set_fibonacci(fibonacci)
    setter.set_square(square)

    setter.do_all(10)

    assert(True)